src/DblParser/Desugar.ml

```diff
@@ -816,4 +837,45 @@ and make_handle ~pos lbl_opt eff_opt pat body hcs =
       fin_clauses = fcs
     })
 
-let tr_program (p : Raw.program) = { p with data = tr_defs p.data }
+and generate_accessor_method_pattern named_type_args type_name =
+  let make data = { data; pos=Position.nowhere } in
+  let create_mapping i arg =
+    let old_name, new_name =
+      match (snd arg.data).data with
+      | TA_Var(name, _) -> name, name ^ "#TA_Var#" ^ string_of_int i
+      | TA_Effect -> "", "#TA_Effect#" ^ string_of_int i
+      | TA_Wildcard -> "", "#TA_Wildcard#" ^ string_of_int i
+    in
+    make (TNVar old_name, make (TA_Var(new_name, make (KWildcard)))), new_name
```

Comment:
For `TA_Effect` and `TA_Wildcard` you generate a type parameter named `""` (`TN_Var ""`), which looks suspicious. I think, it should be `TNEffect` for `TA_Effect` and `TNAnon` for `TA_Wildcard`.